src/beatkeeper.rs

```diff
@@ -469,6 +502,16 @@ impl BeatKeeper {
             }
         }
 
+        // Neue Schleife für die .DAT-Dateipfade
+        for (i, anlz_file) in rb.get_anlz_files()?.iter().enumerate(){
+            // `set()` aktualisiert den Wert nur, wenn er sich geändert hat
+            if self.anlz_files[i].set(anlz_file.clone()){
```

Comment:
I see you followed what I did, but this does not actually have to be cloned. You can instead read the value from the anlz_files object, which tracks changes. Thanks for making me catch this, it's fixed on the dev branch